### PR TITLE
feat(release-notification): update-notification-badge

### LINKED ISSUE #118: Enhance popup usability and design
> **Is your enhancement proposal related to a problem? Please describe.**
> The current popups ("Welcome to the new Phonebook!", "We need your help", "We've updated to a newer Version") can be very annoying if you don't safe cookies.
> 
> **Describe the solution you'd like**
> A popup like "snack-bar" which will open from one site and could be easy closed.
> 
> 
> > Issue #43 can be fixed here

**UPDATE:** The user will see "Welcome-Notification" and "Bug-Report-Dialog" on the first start of the page. If someone don't allow localstorage he will see this dialogs everytime. We should find a way to stop this.
--> There is no way.

## Idea to don't show the dialogs for users with no allowed cookies:
An workaround to don't show the dialogs every time could another url e.g. `phonebook.com/skip_dialogs` which then do not show the welcome-dialog and the major-update dialog. The bug-report-dialog should still be shown.

Probably could easy implement with setting the status before other lifecycle events with https://www.ngxs.io/concepts/actions or by just don't check the values in localstorage on the above url.

**Checklist**
- [x] Provide a new url-parameter (e.g. ?skip_dialog="true")
- [x] If this url is called the welcome popup will not be shown 
- [x] ..the "release-notification-dialog" on the second page load should also not be shown
- [x] Inform about the new url on the information-page and on the welcome-dialog

### PR DESCRIPTION
**Pull request by mschwrdtnr**
**This PR includes PR #131 . So merge this pr after #131** 

image

The user will only get a `displayReleaseDialog` if there is a breaking change. A badge next to the person-icon and the new added mat-menu-item `Changelog` will be shown if there is a new release (breaking, feature, bugfix). The user will also be informed about a feature or bugfix with `displayReleaseNotification`

### CODE REVIEW

**Comment by DanielHabenicht** [comment]
This does not resolve #118 as it only reduces the number of popups for the new version dialogue. The root cause is not resolved.

**Comment by mschwrdtnr** [comment]
> This does not resolve #118 as it only reduces the number of popups for the new version dialogue. The root cause is not resolved.

Then we should talk about what it needs to resolve Issue #118 ...

**Comment by mschwrdtnr** [comment]
@DanielHabenicht is there anything left to do here at this moment?

**Comment by DanielHabenicht** [comment]
Merge please

**Comment by DanielHabenicht** [comment]
Preview Environment ready at https://13.79.237.3

File: .azure/pipelines/Phonebook.preview.yml [hunk 1]
```diff
@@ -211,8 +211,6 @@ jobs:
     displayName: 'Delay Deletion of Preview Environment'
     pool: server
     dependsOn: deploy_preview
```
Comment (DanielHabenicht) [inline .azure/pipelines/Phonebook.preview.yml]: Bitte merge Fehler beheben

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html [hunk 2]
```diff
@@ -125,6 +131,17 @@
...
                 </button>
+                <button mat-menu-item>
+                  <mat-icon>new_releases</mat-icon>
+                  <span
+                    matBadge="!"
+                    matBadgeColor="accent"
+                    matBadgeSize="small"
+                    [matBadgeHidden]="!releaseInfoService.newUpdate"
+                    (click)="releaseInfoService.displayReleaseDialog()"
+                    >Changelog</span
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html]:
```suggestion
                    >Release Notes</span
```

Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html]: also the Translation is missing

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts [hunk 3]
```diff
@@ -40,7 +42,9 @@ export class NavigationComponent implements OnInit, OnDestroy {
     private router: Router,
     public dialog: MatDialog,
     public i18n: I18n,
-    public featureFlagService: FeatureFlagService
+    public featureFlagService: FeatureFlagService,
+    public badge: MatBadgeModule,
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts]: is badge really used?

File: docs/pages/development-guides/general-guide.md [hunk 4]
```diff
@@ -46,7 +46,7 @@ We splitted the development documentation in two different parts.
 **What do you want to start with?**
 
 <span class="fs-6">
-[Frontend](frontend/getting-started){: .btn .btn-purple }
+[Frontend](frontend){: .btn .btn-purple }
```
Comment (DanielHabenicht) [inline docs/pages/development-guides/general-guide.md]: the merge has gone wrong

File: Phonebook.Frontend/src/app/services/release-info.service.ts [hunk 5]
```diff
@@ -61,7 +65,7 @@ export class ReleaseInfoService {
       });
   }
 
-  private displayReleaseNotification() {
+  public displayReleaseNotification() {
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/services/release-info.service.ts]: this function is not used publicly

File: Phonebook.Frontend/src/app/app.component.ts [hunk 6]
```diff
@@ -112,6 +112,7 @@ export class AppComponent implements OnInit {
         height: '90vh',
         width: '90vw'
       });
+      this.releaseMigrationService.newUpdate = true;
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/app.component.ts]: is this for debugging purposes, or anything else?

**Comment by DanielHabenicht** [comment]
Preview Environment ready at https://13.79.237.3

File: Phonebook.Frontend/src/app/app.component.ts (hunk 6 above)
Comment (mschwrdtnr) [reply Phonebook.Frontend/src/app/app.component.ts]: @DanielHabenicht I think that if the application has started for the first time the update-badge should also shown, or not? I'am also not sure^^

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts (hunk 3 above)
Comment (mschwrdtnr) [reply Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts]: It's used at the *person-icon* and at *Release Notes*

File: .azure/pipelines/Phonebook.preview.yml (hunk 1 above)
Comment (mschwrdtnr) [reply .azure/pipelines/Phonebook.preview.yml]: Should be fine now?

**Comment by DanielHabenicht** [comment]
Preview Environment ready at https://13.79.237.3

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts (hunk 3 above)
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts]: but it is not referenced, you only have to import Modules in the import statement of other modules. In *.ts files you only have to import things you are referencing in code, e.g. services

File: Phonebook.Frontend/src/app/app.component.ts (hunk 6 above)
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/app.component.ts]: Well but this case should be covered by the normal update procedure as the default value for version is `0.0.0`

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html [hunk 7]
```diff
@@ -125,6 +131,18 @@
                   <mat-icon>info_outline</mat-icon>
                   <span i18n="@@NavigationComponentInformation">Information</span>
                 </button>
+                <button mat-menu-item>
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html]: This should also be added to the burger menu.

File: .azure/pipelines/Phonebook.preview.yml (hunk 1 above)
Comment (DanielHabenicht) [reply .azure/pipelines/Phonebook.preview.yml]: yep should be fine, although no changes on this file would be even better.

**Comment by DanielHabenicht** [comment]
Preview Environment ready at https://13.79.237.3

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html [hunk 8]
```diff
@@ -125,15 +146,33 @@
...
                 </button>
+                <button mat-menu-item>
+                  <mat-icon>new_releases</mat-icon>
+                  <span
+                    matBadge="!"
+                    matBadgeColor="accent"
+                    matBadgeSize="small"
+                    [matBadgeHidden]="!releaseInfoService.newUpdate"
+                    (click)="releaseInfoService.displayReleaseDialog()"
+                    i18n="NavigationComponent|Release Notes Menu Item@@NavigationComponentReleaseNotes"
```
Comment (DanielHabenicht) [inline Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html]:
```suggestion
                    i18n="@@NavigationComponentReleaseNotes"
```

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts (hunk 3 above)
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.ts]: ?

File: Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html (hunk 8 above)
Comment (DanielHabenicht) [reply Phonebook.Frontend/src/app/shared/components/navigation/navigation.component.html]: @mschwrdtnr just so you know on consecutive translations you can just put the ID of the translation for brevity.

**Pull request merged**